packages/core/contracts-ovm/insured-bridge/implementation/BridgePoolFactory.sol

```diff
@@ -0,0 +1,189 @@
+        _setIdentifier(_identifier);
+    }
+
+    /**
+     * @dev Sets challenge pereiod for relayed deposits. BridgePools will read this value from this
+     * contract. Can only be called by the current owner.
+     */
+    function setOptimisticOracleLiveness(uint256 _liveness) public onlyOwner {
+        _setOptimisticOracleLiveness(_liveness);
+    }
+
+    /**
+     * @dev Sets challenge pereiod for relayed deposits. BridgePools will read this value from this
+     * contract. Can only be called by the current owner.
+     */
+    function setProposerBondPct(uint256 _proposerBondPct) public onlyOwner {
+        _setProposerBondPct(_proposerBondPct);
+    }
+
+    /**
+     * @notice Privileged account can set L2 deposit contract that originates deposit orders to be fulfilled by this
+     * contract.
+     * @dev Only callable by Owner of this contract.
+     * @param _depositContract Address of L2 deposit contract.
+     */
+    function setDepositContract(address _depositContract) public onlyOwner {
+        depositContract = _depositContract;
+        emit SetDepositContract(depositContract);
+    }
+
+    /**
+     * @notice Privileged account can associate a whitelisted token with its linked token address on L2 and its
+     * BridgePool address on this network. The linked L2 token can thereafter be deposited into the Deposit contract
+     * on L2 and relayed via the BridgePool contract.
+     * @dev Only callable by Owner of this contract. Also initiates a cross-chain call to the L2 Deposit contract to
+     * whitelist the token mapping.
+     * @param _l1Token Address of L1 token that can be used to relay L2 token deposits.
+     * @param _l2Token Address of L2 token whose deposits are fulfilled by `_l1Token`.
+     * @param _bridgePool Address of pool contract that stores passive liquidity with which to fulfill deposits.
+     * @param _l2Gas Gas limit to set for relayed message on L2
+     */
+    function whitelistToken(
+        address _l1Token,
+        address _l2Token,
+        address _bridgePool,
+        uint32 _l2Gas
+    ) public onlyOwner {
+        require(_getCollateralWhitelist().isOnWhitelist(address(_l1Token)), "Payment token not whitelisted");
+        // We want to prevent any situation where a token mapping is whitelisted on this contract but not on the
+        // corresponding L2 contract.
+        require(depositContract != address(0), "Deposit contract not set");
+
+        L1TokenRelationships storage whitelistedToken = whitelistedTokens[_l1Token];
+        whitelistedToken.l2Token = _l2Token;
+        sendCrossDomainMessage(
+            depositContract,
+            _l2Gas,
+            abi.encodeWithSignature("whitelistToken(address,address)", _l1Token, whitelistedToken.l2Token)
+        );
+
+        // TODO: This contract should deploy a new BridgePool if the address is set to 0x0 at this point.
+        whitelistedToken.bridgePool = _bridgePool;
```

Comment:
or, if you want to have this act as a two tx process, just validate the correctness of `bridgePool` address

Reply:
Yeah I'm not sure what we should do. On one hand I think its clean to deploy a new contract here and keep it to 1 tx, but the gas costs and code complexity are the other issue. WDYT?

Reply:
I dont think it's too much of an issue, assuming you make the deployment method an internal function that gets called if one is not already deployed. feel free to leave it for a follow on PR though.